themes/fishy.zsh-theme

```diff
@@ -1,13 +1,13 @@
 # ZSH Theme emulating the Fish shell's default prompt.
 
 _fishy_collapsed_wd() {
-  echo $(pwd | perl -pe "
+  echo $(pwd | perl -pe '
    BEGIN {
-      binmode STDIN,  ':encoding(UTF-8)';
-      binmode STDOUT, ':encoding(UTF-8)';
-   }; s|^$HOME|~|g; s|/([^/])[^/]*(?=/)|/\$1|g
-")
-} 
+      binmode STDIN,  ":encoding(UTF-8)";
+      binmode STDOUT, ":encoding(UTF-8)";
+   }; s|^$HOME|~|g; s|/([^/.])[^/]*(?=/)|/$1|g; s|/\.([^/])[^/]*(?=/)|/.$1|g;
```

Comment:
You're right: this is broken! Sorry, and thanks for catching it. I corrected this error in my local copy but failed to commit and push it. I'll do that in a moment.

The reason I changed the quotes is that interpolating shell variables inside a Perl snippet is usually considered a bad practice. Shell and Perl both use `$varname`, though each in a somewhat different way, so it can quickly get confusing. For example, here the `$1` is *not* supposed to be interpolated by the shell (notice the slash escaping the dollar sign before it in the double-quotes version) but `$HOME` *is.* For this reason I think a better solution is to access the value of the environment variable `HOME` through Perl's built in `%ENV` hash.